File: go/encode.go
package js

import (
	"time"
)

func EncodeBool(b bool) []byte {
	bz, err := encodeBool(b)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeByte(b byte) []byte {
	bz, err := encodeByte(b)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeByteSlice(bz []byte) []byte {
	bz2, err := encodeByteSlice(bz)
	if err != nil {
		panic(err)
	}
	return bz2
}

func EncodeFloat32(f float32) []byte {
	bz, err := encodeFloat32(f)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeFloat64(f float64) []byte {
	bz, err := encodeFloat64(f)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeInt16(i int16) []byte {
	bz, err := encodeInt16(i)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeInt32(i int32) []byte {
	bz, err := encodeInt32(i)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeInt64(i int64) []byte {
	bz, err := encodeInt64(i)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeInt8(i int8) []byte {
	bz, err := encodeInt8(i)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeString(s string) []byte {
	bz, err := encodeString(s)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeTime(t time.Time) []byte {
	bz, err := encodeTime(t)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeUint16(u uint16) []byte {
	bz, err := encodeUint16(u)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeUint32(u uint32) []byte {
	bz, err := encodeUint32(u)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeUint64(u uint64) []byte {
	bz, err := encodeUint64(u)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeUint8(u uint8) []byte {
	bz, err := encodeUint8(u)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeUvarint(u uint64) []byte {
	bz, err := encodeUvarint(u)
	if err != nil {
		panic(err)
	}
	return bz
}

func EncodeVarint(i int64) []byte {
	bz, err := encodeVarint(i)
	if err != nil {
		panic(err)
	}
	return bz
}

File: ts/src/encode.ts
import { Amino } from './global';

export function encodeByte (byte: number): Uint8Array {
    return Amino.encodeByte(byte);
}

export function encodeByteSlice (bytes: Uint8Array): Uint8Array {
    return Amino.encodeByteSlice(bytes);
}

export function encodeInt8 (int8: number): Uint8Array {
    return Amino.encodeInt8(int8);
}

export function encodeInt16 (int16: number): Uint8Array {
    return Amino.encodeInt16(int16);
}

export function encodeInt32 (int32: number): Uint8Array {
    return Amino.encodeInt32(int32);
}

export function encodeInt64 (int64: number): Uint8Array {
    return Amino.encodeInt64(int64);
}

export function encodeVarint (varint: number): Uint8Array {
    return Amino.encodeVarint(varint);
}

export function encodeUint8 (uint8: number): Uint8Array {
    return Amino.encodeUint8(uint8);
}

export function encodeUint16 (uint16: number): Uint8Array {
    return Amino.encodeUint16(uint16);
}

export function encodeUint32 (uint32: number): Uint8Array {
    return Amino.encodeUint32(uint32);
}

export function encodeUint64 (uint64: number): Uint8Array {
    return Amino.encodeUint64(uint64);
}

export function encodeUvarint (uvarint: number): Uint8Array {
    return Amino.encodeUvarint(uvarint);
}

export function encodeFloat32 (float: number): Uint8Array {
    return Amino.encodeFloat32(float);
}

export function encodeFloat64 (float: number): Uint8Array {
    return Amino.encodeFloat64(float);
}

export function encodeBool (bool: boolean): Uint8Array {
    return Amino.encodeBool(bool);
}

export function encodeString (string: string): Uint8Array {
    return Amino.encodeString(string);
}

export function encodeTime (time: Date): Uint8Array {
    return Amino.encodeTime(time);
}
